packages/next/src/shared/lib/router/router.ts

```diff
@@ -1035,6 +1067,32 @@ export default class Router implements BaseRouter {
     // hydration. Your app should _never_ use this property. It may change at
     // any time without notice.
     const isQueryUpdating = (options as any)._h === 1
+
+    if (process.env.__NEXT_CLIENT_ROUTER_FILTER_ENABLED) {
+      const asNoSlash = removeTrailingSlash(as)
+      const matchesBflStatic = this._bfl_s?.has(asNoSlash)
+      let matchesBflDynamic = false
+      const asNoSlashParts = asNoSlash.split('/')
+
+      // if any sub-path of as matches a dynamic filter path
+      // it should be hard navigated
+      for (let i = 0; i < asNoSlashParts.length + 1; i++) {
+        const currentPart = asNoSlashParts.slice(0, i).join('/')
+        console.log('checking for', currentPart)
```

Comment:
```suggestion
```

Reply:
removed in https://github.com/vercel/next.js/pull/46287/commits/11f3983efae6399ed59959ff252a020e076148d2